mne/viz/decoding.py

```diff
@@ -129,16 +147,49 @@ def plot_gat_diagonal(gat, title=None, xmin=None, xmax=None, ymin=0., ymax=1.,
     import matplotlib.pyplot as plt
     if ax is None:
         fig, ax = plt.subplots(1, 1)
-    # detect whether gat is a full matrix or just its diagonal
+
+    # Detect whether gat is a full matrix or just its diagonal
     if np.all(np.unique([len(t) for t in gat.test_times_['times_']]) == 1):
         scores = gat.scores_
+    elif train_time == 'diagonal':
+        # Get scores from identical training and testing times even if GAT
+        # is not square.
+        scores = np.zeros(len(gat.scores_))
+        for i, train_time in enumerate(gat.train_times['times_']):
+            for test_times in gat.test_times_['times_']:
+                # find closest testing time from train_time
+                j = np.abs(test_times - train_time).argmin()
+                # check that not more than 1 classifier away
+                if train_time - test_times[j] <= gat.train_times['step']:
+                    scores[i] = gat.scores_[i][j]
+    elif type(train_time) in [float, np.float64, np.float32]:
+        train_times = gat.train_times['times_']
+        idx = np.abs(train_times - train_time).argmin()
+        if train_times[idx] - train_time > gat.train_times['step']:
+            raise ValueError("No classifier trained at %s " % train_time)
+        scores = gat.scores_[idx]
     else:
-        scores = np.diag(gat.scores_)
+        raise ValueError("train_time must be \'diagonal\' or a float.")
     ax.plot(gat.train_times['times_'], scores, color=color,
             label="Classif. score")
-    ax.axhline(0.5, color='k', linestyle='--', label="Chance level")
+    # Find chance level
+    if chance is True:
+        # XXX JRK This should probably be solved within sklearn?
+        if gat.scorer_.__name__ is 'accuracy_score':
+            chance = 1. / len(np.unique(gat.y_train_))
+        elif gat.scorer_.__name__ is 'roc_auc_score':
```

Comment:
we should not rely on this. bad idea to access the double underscore attribute. We should find another way. Second, don't use `is` for a string comparison